README.md

```diff
@@ -69,6 +69,21 @@ In this example,
 
 The characters `"@"` and `"#"` identify the origin and the code, respectively. (We can always introduce additional prefix characters in the future if it turns out we need to include additional information in these messages.)
 
+Some sites use third-party `iframe`s for authentication. In such cases, the third-party `iframe`'s origin can be specified using a `"%"` field after the code.
+
+```text
+747723 is your ExampleCo authentication code.
+
+@example.com #747723 %ecommerce.example
+```
+
+In this example,
+
+* `"https://example.com"` is the origin the code is associated with,
+* `"747723"` is the code,
+* `"https://ecommerce.example"` is the origin of the third-party `iframe`, and
```

Comment:
The attack vector I'm concerned about here is something along the lines of an ad network running on an iframe and sending SMSes to users to identify them. There is user permission involved, which I think helps, but still feels more dangerous than needed.

Can you help me understand what are the alternatives you considered?

Reply:
I’m not sure I’m seeing the attack. The iframe extension only lets the party sending the text bind the code to an iframe they trust on their own page. Can you be more specific about the attack vector here?

The alternatives considered were:
1. Implicitly allow a code bound to a website to be used on all subframes of the website. We considered this to be a potential security risk.
2. Never allow use of origin-bound codes in iframes. We found this to be limiting, because some real-world websites delegate authentication to use auth widgets from other origins.

Reply:
The attack vector I have in mind here is that there are a lot of widgets that are embedded on websites as iframes at scale. Ad networks, analytics, commenting widgets, etc are highly deployed. 

1. highly-deployed iframe-widget evil.com wants to find a user on the web on example.com
1. example.com embeds evil.com as an iframe
1. user goes to example.com
1. evil.com suspects a specific user of interest is at example.com and sends an SMS to the user's phone
1. browser shares the SMS with evil.com while the user is on example.com
1. evil.com now gets confirmation that the specific user was visiting example.com

This, I think, is mitigated if the user agent asks for the user's consent on step (5), which I think is sufficiently true for the autocomplete="one-time-code" formulation which uses autocomplete and the current formulation of WebOTP, but seems like a hard requirement going forward for future implementations of WebOTP (at least for iframe-based use cases it would seem). I'm wondering if this requirement needs to be encapsulated here in this spec or on the WebOTP spec specifically.

Reply:
Other specifications have resolved this class of issue through [Feature Policy](https://w3c.github.io/webappsec-feature-policy/). If usage of WebOTP in cross-origin frames required opt-in from the top-level origin then I think that would resolve the concern in @rmondello's alternative (1).